### PR TITLE
Feature: support SLB to expose raven service

### LINKED ISSUE #22: [feature request] support SLB as public network exporter for gateway
As we known, SLB is very common on cloud nodes that used by end user to expose public service, so if we can support SLB for gateway, then end user can use SLB to expose gateway on cloud nodes to gateways on edge nodes.

### PR DESCRIPTION
**Pull request by njucjc**

### CODE REVIEW

**Comment by rambohe-ch** [comment]
@njucjc maybe as you known, controllers in this repo(raven-controller-manager) is under moving into `yurt-manager` component in `openyurtio/openyurt` repo. so how about post this pull request into `yurt-manager` component?

File: pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go [hunk 1]
```diff
@@ -39,6 +51,8 @@ type GatewaySpec struct {
 	// TODO add a field to configure using vxlan or host-gw for inner gateway communication?
 	// Endpoints is a list of available Endpoint.
 	Endpoints []Endpoint `json:"endpoints"`
+	// Type determines how the Gateway is exposed.
+	Type ExposeType `json:"type,omitempty"`
```
Comment (BSWANG) [inline pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go]: Is it better named `ExposeType`? or the `type` field on CRD will be ambiguity

**Review by BSWANG — CHANGES_REQUESTED** [review]

File: pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go (hunk 1 above)
Comment (njucjc) [reply pkg/ravencontroller/apis/raven/v1alpha1/gateway_types.go]: done

**Comment by njucjc** [comment]
> @rambohe-ch I suggest that we can merge this PR first, and then moving it into `yurt-manager` together.

**Comment by rambohe-ch** [comment]
> > @rambohe-ch I suggest that we can merge this PR first, and then moving it into `yurt-manager` together.

@njucjc ok, i have no problem. and would you be able to migrate this controller into yurt-manager?

**Pull request merged**